### ISSUE TITLE
Billing UX Improvements

### ISSUE THREAD

**Issue by travjenkins** [estuary/ui#1405]
**Last Updated:** Jul. 16, 2026

<hr />

There are a few UX issues with billing that _should_ be supported by Stripe and our services today.

Below are the base requirements for this feature:

*  **`PRD Item 2:`** Users can see all their receipts.

*  **`PRD Item 3:`** When an invoice is paid we should show a receipt.

Stretch Goals:

- **`PRD Item 1:`** Allow users to provide multiple email addresses and allow editing of emails.

-  **`PRD Item 9:`** Show top task usage on the billing page in chart. 

-  **`PRD Item 14:`** Allow "by prefix" break down of usage on billing page.  Eg. if I have access to multiple tenants, I can select one and see its billing page

## Dependencies

**Last Updated:** Apr. 23, 2026

- [x] [GraphQL.](https://github.com/estuary/flow/issues/2879) _Migrate the billing-related, Supabase edge functions to GraphQL._

## References

**Requirements**

* [Billing Updates PRD.](https://docs.google.com/document/d/1UPU5Fyfivcy9AI1bl-DR5wnIOPvbB9ooAiDgIs6EzSI/edit?tab=t.0) _A document identifying a series of UX improvements for the Billing tab of the Admin page._

**Comment by travjenkins** [comment]
Stripe mainly supports receipts through emails and opening a link to their site.
https://docs.stripe.com/receipts

They do have an API that provides receipts for users
https://docs.stripe.com/api/charges/object#charge_object-payment_method_details-card_present-receipt

We might be able to offer PDF download of the receipt (like we do with invoices). Or we could allow users to ask for an email when something is paid?

**Comment by kiahna-tucker** [comment]
This feature is blocked by a local development issue concerning the billing edge function. The edge function is throwing an error (with `code: "WORKER_ERROR"`) that prevents the _Payment Information_ section of the _Billing_ tab on the _Admin_ page from rendering. The error was first reported via Slack on Apr. 3rd.

**Comment by travjenkins** [comment]
Can we split off the stretch goals into a stand alone PR since the move to GQL might take awhile?

**Comment by kiahna-tucker** [comment]
That depends on how we are defining _usage_ and where we would like to pull data from long-term. Presently, invoices are treated as the source of truth for data volume and the two metrics which are encapsulated by the term _task usage_. The task-specific graph that previously existed in the client extracted the task hours metric from invoices.
